lib/Backend/GlobOpt.cpp

```diff
@@ -13477,8 +13477,11 @@ GlobOpt::TypeSpecializeStElem(IR::Instr ** pInstr, Value *src1Val, Value **pDstV
     case ObjectType::Float64VirtualArray:
     case ObjectType::Float64MixedArray:
     Float64Array:
-        toType = TyFloat64;
-        break;
+    if (this->DoFloatTypeSpec())
```

Comment:
We also need to disable all of the int cases if DoAggressiveIntTypeSpec or DoFloatTypeSpec is disabled, like I did in ValueNumberLdElemDst()